pkg/asset/installconfig/gcp/validation.go

```diff
@@ -141,66 +141,144 @@ func validateZoneProjects(client API, ic *types.InstallConfig, fieldPath *field.
 	return allErrs
 }
 
-// validateManagedZones will validate the public and private managed zones if they exist.
-func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
-	commonZoneLookup := func(dnsZone, zoneProject, defaultProject string) error {
-		project := zoneProject
-		if project == "" {
-			project = defaultProject
-		}
+// internalFindProject finds the correct project to use during installation. If the project id is
+// provided in the zone use the project id, otherwise use the default project.
+func internalFindProject(zone *gcp.DNSZone, defaultProject string) string {
+	if zone != nil && zone.ProjectID != "" {
+		return zone.ProjectID
+	}
+	return defaultProject
+}
```

Comment:
I think this is fine, because I know this pattern is being used a lot from CORS-2040. But I would suggest an alternative which you could consider, which is to [set a default](https://github.com/openshift/installer/tree/master/pkg/types/gcp/defaults), so if `zone != nil && zone.ProjectID == ""` you set `zone.ProjectID` to the project used for auth. Then in all subsequent calls to `zone.ProjectID` you know it will be populated.